src/daemon/daemon.c

```diff
@@ -33,6 +33,7 @@
 #include <glib-object.h>
 #include <gtk/gtk.h>
 
+#ifdef HAVE_X11
```

Comment:
You can also check that the backend is available, not just enabled:
- enabled `#ifdef HAVE_X11`
- available `#ifdef GDK_WINDOWING_X11`

Both things:
- enabled & available `#if defined(HAVE_X11) && defined(GDK_WINDOWING_X11)`

Reply:
That would be inconvenient to put everywhere, and is almost certainly not necessary. Only GTK versions *much* older than the required v3.22 don't support Wayland, and I can't imagine distro's patching it out. If that did become an issue it should be up to the build system to turn off `HAVE_X11` if GTK doesn't support it.

Reply:
GTK can be shipped without X11 backend support:
```
AC_ARG_ENABLE(x11-backend,
              [AS_HELP_STRING([--enable-x11-backend],
                              [enable the X11 gdk backend])],
			      [backend_set=yes])
```
https://github.com/GNOME/gtk/blob/1d2e0b7e5c6350ae412c217b7a524cb6d3686a2c/configure.ac#L325-L328